next-tavla/app/(admin)/edit/[id]/components/MetaSettings/actions.ts

```diff
@@ -64,3 +65,67 @@ async function getTilesWithDistance(board: TBoard, location?: TLocation) {
         }),
     )
 }
+
+export async function moveBoardToPersonal(
+    bid: TBoardID,
+    from?: TOrganizationID,
+) {
+    const user = await getUserFromSessionCookie()
+    if (!user) return getFormFeedbackForError('auth/operation-not-allowed')
+
+    if (from) {
+        const orgAccess = await hasBoardEditorAccess(bid)
+        if (!orgAccess) return redirect('/')
+    }
+
+    firestore()
+        .collection('users')
+        .doc(String(user.uid))
+        .update({
+            ['owner']: admin.firestore.FieldValue.arrayUnion(bid),
+        })
+
+    firestore()
+        .collection(from ? 'organizations' : 'users')
+        .doc(from ? String(from) : String(user.uid))
+        .update({
+            [from ? 'boards' : 'owner']:
+                admin.firestore.FieldValue.arrayRemove(bid),
+        })
+
+    revalidatePath(`/edit/${bid}`)
+}
+
+export async function moveBoardToOrganization(
+    bid: TBoardID,
+    oid?: TOrganizationID,
+    from?: TOrganizationID,
+) {
+    const user = await getUserFromSessionCookie()
+    if (!user) return getFormFeedbackForError('auth/operation-not-allowed')
+
+    const orgAccess = await hasBoardEditorAccess(bid)
+    if (!orgAccess) return redirect('/')
+
+    if (from) {
+        const orgAccess = await hasBoardEditorAccess(bid)
+        if (!orgAccess) return redirect('/')
+    }
```

Comment:
Hva skjer hvis from er undefined

Reply:
Hvis from er undefined er det fra en personlig tavle, og da trengs det ikke å sjekke for organisasjonstilgang.